### PR TITLE
Update quantization round and clip calculation rules

### PR DESCRIPTION
**Pull request by yghstill**
### PR types

Function optimization 
### PR changes

APIs
### Describe

Update quantization round and clip calculation rules:
- The round method support `rounding to nearest ties to even` and `rounding to nearest ties away from zero`.  And add new attr `round_type`.
- INT8 value range changed from [-127, 127] to [-128, 127].
- clip functor after round.

### CODE REVIEW

File: python/paddle/fluid/contrib/slim/tests/test_imperative_ptq.py [hunk 1]
```diff
@@ -338,7 +338,7 @@ def set_vars(self):
 
         self.batch_num = 10
         self.batch_size = 10
-        self.eval_acc_top1 = 1.0
+        self.eval_acc_top1 = 0.98
```
Comment (ceci3) [inline python/paddle/fluid/contrib/slim/tests/test_imperative_ptq.py:341]: 这里是精度下降了才改的吗？

Comment (yghstill) [reply python/paddle/fluid/contrib/slim/tests/test_imperative_ptq.py:341]: 精度有波动，acc_top1调整和其他单测保持一致

**Review by wanghaoshuang — APPROVED** [review]
LGTM

**Review by ceci3 — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]
LGTM for 
`set_tests_properties(test_quant2_int8_resnet50_mkldnn PROPERTIES TIMEOUT 200)`

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by cyj1986 — APPROVED** [review]
LGTM

**Pull request merged**